datastream/dataset.py

```diff
@@ -358,3 +434,38 @@ def test_combine_dataset():
         )
         for index, inner_indices in enumerate(indices)
     )
+   
+
+def test_split_dataset():
+    dataset = Dataset.from_dataframe(pd.DataFrame(dict(
+        index=np.arange(100),
+        number=np.random.randn(100),
+        stratify=np.concatenate([np.ones(50), np.zeros(50)]),
+    ))).map(tuple)
+
+    split_file = Path('test_split_dataset.json')
```

Comment:
you can use the `io.StringIO` to avoid using files
https://docs.python.org/3/library/io.html

Reply:
Yes I figured but didn't want to put time into learning how to yet. Also I don't think StringIO will be easy to understand in the documentation so would be nice to hide the code or something for the user. That's probably possible as well but also something I didn't want to put time into learning (yet at least).

I think I saw some pytest solution to mocking files. Maybe that's a better solution

Reply:
Looked into StringIO. It does not work here because we take a pathlib.Path and not a file object. Haven't looked into mocking files, that would probably work. I don't think we have to change this now.